docs-md/concepts/materialized-views.md

```diff
@@ -1,16 +1,53 @@
+Streams and tables are closely related. A stream is a sequence of events that
+you can derive a table from. For example, a sequence of credit scores for a
+loan applicant can change over time. The sequence of credit scores is a stream.
+But this stream be interpreted as a table to describe the applicant's current
+credit score.
 
-For our purposes, the important thing that enables efficient queries is access to the current state of applied changes. This is more commonly known as a materialized view. 
+Conversely, the table that represents current credit stores is really two
+things: the current credit scores, and also the sequence of changes to the
+credit scores for each applicant. This is a profound realization, and much has
+been written on this stream/table duality.  Many databases throw away the
+series of changes to a table and only keep the current state around, which
+breaks the natural duality.
 
-More precisely, a materialized view is the result of incrementally-updated aggregation over a stream. When a new event is integrated, the current state of the view evolves into a new state. This transition happens by applying the aggregation function that defines the view with the current state and the new event. In this way, a view is never “fully recomputed” when new events arrive. It simply incrementally adjusts to account for the new information. Thus, queries against materialized views are highly efficient because they simply look up the value of the current state. They do no further work.
+What enables efficient queries is access to the current state of applied
+changes. This is more commonly known as a *materialized view*. 
 
-In ksqlDB, a table can either be materialized into a view or not. If a table is created directly on top of a Kafka topic, it is not materialized. Non-materialized tables cannot be queried because they would be highly inefficient. On the other hand, if a table is derived from another collection, ksqlDB will materialize its results, and queries can be made against it.
+A materialized view is the result of incrementally-updated aggregation over
+a stream. When a new event is integrated, the current state of the view evolves
+into a new state. This transition happens by applying the aggregation function
+that defines the view with the current state and the new event. In this way,

```

Comment:
When a new event is integrated, the current state of the view evolves
into a new state. This transition happens by applying the aggregation function
that defines the view with the current state and the new event. 

When a new event is integrated the aggregation function
that defines the view is applied only on this new event leading to a new state for the view.